### ISSUE TITLE
`spyOn().mockReturnValue()` does not work with module namespace objects

### ISSUE THREAD

**Issue by callumfrance** [oven-sh/bun#6677]
### What version of Bun is running?

1.0.6

### What platform is your computer?

Darwin 23.0.0 arm64 arm

### What steps can reproduce the bug?

Example file `trials.ts`:

```ts
export function foobar() {
    console.log("I am foobar real");
    return 1;
}

export function helloWorld() {
    console.log("I am hello world real");
    return foobar();
}
```

Example test file `trials.test.ts`:

```ts
import { expect, spyOn, test } from "bun:test";

import * as trials from "./trials";

test("random jesty", async () => {
    spyOn(trials, "foobar").mockReturnValue(3);
    const res = trials.helloWorld();

    expect(res).toEqual(3);
});
```

Executed on command line - `bun test src/trials.test.ts`

### What is the expected behavior?

The test should pass successfully as the mocked return value of `foobar()` is `3`.

### What do you see instead?

The test does not pass as the actual, un-mocked implementation of `foobar()` is used as it returns `1` -

```
$ bun test src/trials.test.ts
bun test v1.0.6 (969da088)

src/trials.test.ts:
I am hello world real
I am foobar real
4 |
5 | test("random jesty", async () => {
6 |     spyOn(trials, "foobar").mockReturnValue(3);
7 |     const res = trials.helloWorld();
8 |
9 |     expect(res).toEqual(3);
        ^
error: expect(received).toEqual(expected)

Expected: 3
Received: 1

      at /***/src/trials.test.ts:9:4
      at /***/src/trials.test.ts:5:21
✗ random jesty [0.40ms]

 0 pass
 1 fail
 1 expect() calls
Ran 1 tests across 1 files. [15.00ms]
```

### Additional information

This is an attempt at a workaround within `bun:test` whilst https://github.com/oven-sh/bun/issues/5394 remains unsolved.

**Comment by Jarred-Sumner** [comment]
This is due to the ESM namespace object specifically - they need to use a special code path to make the properties mutable

**Comment by Jarred-Sumner** [comment]
Fixed in #6685
